### PR TITLE
Create deploy.yml

### PR DESCRIPTION
**Pull request by praeclarum**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# For help, go to https://github.com/Azure/Actions
+#
+# 1. Set up the following secrets in your repository:
+#   AZURE_WEBAPP_PUBLISH_PROFILE
+#
+# 2. Change these variables for your configuration:
+env:
+  AZURE_WEBAPP_NAME: fuget    # set this to your application's name
+  AZURE_WEBAPP_PACKAGE_PATH: '.'      # set this to the path to your web app project, defaults to the repository root
+  DOTNET_VERSION: '2.2.402'           # set this to the dot net version to use
```
Comment (timheuer) [inline .github/workflows/deploy.yml:20]: 2.2 is out of support FYI

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    environment: dev
+    steps:
+
+      # Checkout the repo
+      - uses: actions/checkout@master
```
Comment (timheuer) [inline .github/workflows/deploy.yml:29]: Best practice in Actions is not to use @master versioning -- use a specific version, in this case checkout is at @v2.3.4

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      
+      # Setup .NET Core SDK
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
```
Comment (timheuer) [inline .github/workflows/deploy.yml:33]: Move to @v1.8.0

**Review by timheuer — COMMENTED** [review]
Some suggestions

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+name: Deploy to Azure Web App
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
[the remark this answers is not in the document]
Comment (praeclarum) [reply .github/workflows/deploy.yml:7]: Yeah I was just cheating here temporarily :-)

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (praeclarum) [reply .github/workflows/deploy.yml:20]: Yeah my goal is to get up to .NET 5, but I don't want to deal with any code breaks right now. Or are you saying deployments will fail?

**Pull request merged**